src-js/analytics.js

```diff
@@ -0,0 +1,56 @@
+class AnalyticsClient {
+  static getSessionId() {
+    var existingSessionId = Number(Cookies.get("amplitude-session-id"));
+
+    // Number.isNaN polyfill:
+    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN
+    const isNaN = Number.isNaN || ((value) => value !== value);
+    return !isNaN(existingSessionId) ? existingSessionId : Date.now();
+  }
+
+  /**
+   * Thin wrapper function around Segment Analytics track method
+   * @param name  Name of the action the current user performed.
+   * @param properties  A bag of metadata to send to Segment/Amplitude.
+   */
```

Comment:
Thank you for doing docstrings 🎉 !

Reply:
All credits goes to: https://github.com/circleci/frontplane/blob/main/packages/analytics/src/analytics.ts 😂